vtex/faststore
# feat: Next 13 global fonts

## What's the purpose of this pull request?

This PR intends to introduce `next/font` built-in feature to the project (`app` directory).

The `babelrc.js` config file needed to be removed in favor of `next/font` usage, [it's mandatory](https://nextjs.org/docs/messages/babel-font-loader-conflict). Next.js requires the use of its own compiler (SWC) config, so for now we still don't have the necessary GraphQL Codegen plugin to replace the `@graphql-codegen/client-preset` being loaded in the Babel config file, as the existing one ([`@graphql-codegen/client-preset-swc-plugin`](https://the-guild.dev/graphql/codegen/plugins/presets/preset-client#swc-plugin)) has a bug that hasn't been solved yet (see [here](https://github.com/dotansimha/graphql-code-generator/issues/9753) and [here](https://github.com/dotansimha/graphql-code-generator/issues/9450).

Also, right now we are just introducing the feature, but we will have to support local fonts and font override later in other tasks.

## How to test it?

- Run the local server (`yarn turbo run dev --filter=@faststore/core`);
- Make sure that all the current pages are working as expected (`/`, `/office` and some `/<slug>/p`);
- Make sure `/fs-next-update` is rendering with some text on the top of the page and without errors.

## References

- https://nextjs.org/blog/next-13#nextfont
- https://nextjs.org/docs/app/api-reference/components/font#css-variables
- https://nextjs.org/docs/app/building-your-application/optimizing/fonts
